server/src/main/java/org/opensearch/index/mapper/ObjectMapper.java

```diff
@@ -380,6 +380,9 @@ protected static void parseProperties(ObjectMapper.Builder objBuilder, Map<Strin
                         throw new MapperParsingException("No handler for type [" + type + "] declared on field [" + fieldName + "]");
                     }
                     String[] fieldNameParts = fieldName.split("\\.");
+                    if (fieldNameParts.length < 1) {
+                        throw new MapperParsingException("Index -1 out of bounds for length 0");
```

Comment:
Two bits of feedback here - 

1. It isn't immediately obvious when this if-clause is satisfied, so please add a comment like

```
// field name is just ".", which is invalid
```

2. Since we return a HTTP 400 error, this message doesn't really tell the customer/client what action to take to fix their request. I would suggest an error message like:

```
"Invalid field name" + fieldName
```